Apromore-Core-Components/Apromore-Portal/src/main/java/org/apromore/portal/dialogController/PopupMenuController.java

```diff
@@ -399,81 +407,75 @@ private void viewExistingDashboard() {
 
     }
 
-    private void addSubMenuItem(Menupopup popup, String subMenuId) {
+    private void addProcessDiscoverSubMenuItem(Menupopup popup) {
         try {
-            String subMenuImagePath = getSubMenuImage(subMenuId);
-            if (subMenuImagePath == null) {
+            Set<Object> selections = getSelections();
+            if (selections.isEmpty()) {
                 return;
             }
-            MenuConfig menuConfig = menuConfigLoader.getMenuConfig(subMenuId);
-            if (menuConfig.getGroups().isEmpty()) {
+            if (pluginAvailable(PluginCatalog.PLUGIN_DISCOVER_MODEL)) {
+                new DiscoverPopupLogSubMenuController(this, mainController, popup,
+                    (LogSummaryType) selections.iterator().next());
+            }
+            } catch (Exception e) {
+            LOGGER.error(FAILED_LOAD_SUB_MENU, e);
+        }
+    }
```

Comment:
There's a chunk of repeated code in here common with `addLogFilterSubMenuItem`, `addDashboardSubMenuItem` etc.  There's probably a clever way to pass the constructor as an argument, specify the construct as a generic, or move it to one of the subclasses, but I think it's tricky enough that I'm willing to let it slide unless you're ambitious to figure it out.